### PR TITLE
41 request interests crud

### LINKED ISSUE #41: Add CRUD functionality for request interest
Using [Django REST framework](https://www.django-rest-framework.org/), create a GET, POST, DELETE/PUT/PATCH endpoints to query request interest. 
```
GET /interests/                                       (Get all interests made on the app)
GET /requests/<id>/interests/               (Get all interest for a particular mentorship request)
GET /interests/<id>                               (Get 1 particular interest)
POST /interests/                                    (Create an interest)
PUT/PATCH /interests/<id>
DELETE /interests/<id>
```

### LINKED ISSUE #84: Write tests for interests view
write tests for views/interest.py

### PR DESCRIPTION
**Pull request by Nkoli**
Fixes #41
Fixes #84

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -161,8 +161,7 @@ def __str__(self):
 
 class RequestInterest(models.Model):
     request = models.ForeignKey(Request, on_delete=models.CASCADE)
-    user = models.ForeignKey(User, on_delete=models.CASCADE,
-                             related_name='mentor'),
+    mentor = models.ForeignKey(User, on_delete=models.CASCADE, default='')
```
Comment (Del-sama) [inline app/models.py]: Why did you change this field name and why is there a default?

File: app/oauth.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import os
 import re
-
 import requests
-from django.contrib.auth.models import User as DefaultUser
+
 from dotenv import load_dotenv
 from oauth2_provider.models import AccessToken
 
+from django.contrib.auth.models import User as DefaultUser
```
Comment (Del-sama) [inline app/oauth.py]: I think this should remain about the dotenv import

File: app/urls.py [hunk 3]
```diff
@@ -81,5 +81,17 @@
...
+    path('interests/',
+         RequestInterestList.as_view(),
+         name='request_interest_list'
+         ),
+    path('interests/<int:pk>/',
+         RequestInterestDetail.as_view(),
+         name='request_interest_detail'
+         ),
+    path('requests/<int:pk>/interests/',
+         MentorRequestInterest.as_view(),
```
Comment (Del-sama) [inline app/urls.py]: I'm not crazy about these namings, I'll probably update them in my refactor pull request

File: app/views/requests_interests.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+class RequestInterestList(generics.ListCreateAPIView):
+    permission_classes = (IsAuthenticated,)
+    queryset = RequestInterest.objects.all()
+    serializer_class = RequestInterestSerializer
+
+
+class MentorRequestInterest(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get_object(self, pk):
```
Comment (Del-sama) [inline app/views/requests_interests.py:30]: I don't know if `get_object` is a good name here, being that we're getting a list of interests

File: app/views/requests_interests.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        try:
+            return RequestInterest.objects.filter(request_id__exact=pk)
+        except RequestInterest.DoesNotExist:
+            raise Http404
+
+    def get(self, request, pk):
+        interest = self.get_object(pk)
```
Comment (Del-sama) [inline app/views/requests_interests.py]: `Interests` maybe

File: app/views/requests_interests.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        serializer = RequestInterestSerializer(interest, many=True)
+        return Response(serializer.data)
+
+
+class RequestInterestDetail(generics.RetrieveUpdateDestroyAPIView):
```
Comment (Del-sama) [inline app/views/requests_interests.py:42]: Shouldn't this get a single `RequestInterest`? Why aren't we passing in a primary key and doing a get instead of a get all

**Review by Del-sama — COMMENTED** [review]
See comments

**Comment by Del-sama** [comment]
I'm going to open a pull request today that will require docstrings for classes, methods and functions, might as well start adding docstrings now so we don't have to come back to these

File: app/models.py (hunk 1 above)
Comment (Nkoli) [reply app/models.py]: We discussed this earlier on [here](https://python-mentorship.slack.com/archives/CPN1K9MB9/p1597325169005000), Tolu said having the name as 'user' was resulting in an error and changing the name to 'mentor' was the only way she could fix it. I added the default empty object because the field already existed in my db and I didn't want to drop the db, using the 'timezone.now' default doesn't work because of the field type.

File: app/oauth.py (hunk 2 above)
Comment (Nkoli) [reply app/oauth.py]: The PEP8 styling guide says something else, I had it above dotenv before i reordered it because of the previous comment Kosy left alongside the style guide link.

File: app/urls.py (hunk 3 above)
Comment (Nkoli) [reply app/urls.py]: lol okay no wahala

File: app/views/requests_interests.py (hunk 5 above)
Comment (Nkoli) [reply app/views/requests_interests.py]: okay, noted

File: app/views/requests_interests.py (hunk 6 above)
Comment (Nkoli) [reply app/views/requests_interests.py:42]: the generics class-based view is a shorter version of APIView, does the exact same thing as well. Also, with the APIView, I was unable to run the put or patch operation, so i switched to this. It gets a single requestinterest by the way, the retrieve part handles that.

**Review by kosyfrances — APPROVED** [review]
